### PR TITLE
[Experimental]: Update: new title and icon for the Overlay Navigation block

### LINKED ISSUE #49980: Product Filters Redesign: Rename Navigation blocks
## Description

The navigation block has the same name and icon as the Core Navigation block, but it only opens/closes the overlay. It may be confusing to merchants. Can we change it to other name and use a different icons to distinguish from the Core one?

### PR DESCRIPTION
**Pull request by dinhtungdu**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This block updates the title and icon of the Overlay Navigation block according to [new design](https://github.com/woocommerce/woocommerce/issues/49980#issuecomment-2250147779) provided by Jarek.

Closes #49980.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build` command to have experimental blocks available.
1. Edit Product Filters Overlay template part.
2. Select the Navigation block.
3. See the new icon and title match the design below.

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/changelog/fix-49980-overlay-navigation-title-icon [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: update
+
+Update: New title and icon for Overlay Navigation block
```
Comment (roykho) [inline plugins/woocommerce/changelog/fix-49980-overlay-navigation-title-icon]: I think we just need a comment here so the changelog doesn't make it into the official list.

Comment (dinhtungdu) [reply plugins/woocommerce/changelog/fix-49980-overlay-navigation-title-icon]: Good call! I updated the changelog to comment. Thanks!

**Review by roykho — APPROVED** [review]
Thanks for working on this. Changes looks good to me. :ship:

✅ The title and icon has been updated.

**Review by thealexandrelara — APPROVED** [review]
Thank you for working on this. Code looks good, and the title and icon matches the design. Nice touch on organizing the exports for the icons in alphabetic order 🙌 .

**Pull request merged**